taxi/views.py

```diff
@@ -87,3 +89,36 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Driver
+    form_class = DriverCreationForm
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverLicenceUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+
+
+def add_driver_to_car(request):
+    car_id = request.POST.get("car_id")
+    driver = request.user
+
+    Car.objects.get(pk=car_id).drivers.add(driver)
+
+    return redirect("taxi:car-detail", pk=car_id)
+
+
+def remove_driver_from_car(request):
+    car_id = request.POST.get("car_id")
+    driver = request.user
+
+    Car.objects.get(pk=car_id).drivers.remove(driver)
+
+    return redirect("taxi:car-detail", pk=car_id)
```

Comment:
it looks pretty similar, could u make it with only one view ?

Reply:
but don't we break the principle of singe responsibility ?

Reply:
@panicua nope, we don't because REST says that one url e.g. users/ should support GET(list) and POST methods and users/{id}/ can support get, patch, put delete. In this case we don't break that principle